src/components/panel/panel.js

```diff
@@ -988,6 +1007,48 @@ MdPanelRef.prototype.focusOnOpen = function() {
 
 
 /**
+ * Shows the backdrop.
+ */
+MdPanelRef.prototype._showBackdrop = function() {
+  if (this._config.hasBackdrop) {
+    this._backdrop = this._$mdUtil.createBackdrop(this._$rootScope,
+        "_md-panel-backdrop md-opaque");
+    this._$animate.enter(this._backdrop, getElement(this._config['attachTo']));
+  }
+};
+
+
+/**
+ * Hides the backdrop.
+ * @param {boolean} destroy
+ */
+MdPanelRef.prototype._hideBackdrop = function(destroy) {
+  if (this._backdrop) {
+    destroy ? this._backdrop.remove() : this._$animate.leave(this._backdrop);
+  }
+};
+
+
+/**
+ * Ensure the dialog container fill-stretches to the viewport
+ */
+MdPanelRef.prototype._stretchPanelContainerToViewport = function() {
+  var isFixed = this._$window.getComputedStyle(this._$document[0].body)
```

Comment:
This function could definitely use some comments to explain what/why it's doing.